pkg/tnf/handlers/crdstatus/crdstatus.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2020-2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package crdstatus
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/test-network-function/pkg/tnf"
+	"github.com/test-network-function/test-network-function/pkg/tnf/dependencies"
+	"github.com/test-network-function/test-network-function/pkg/tnf/identifier"
+	"github.com/test-network-function/test-network-function/pkg/tnf/reel"
+)
+
+const (
+	ocLabelQueryFlag = " -l "
```

Comment:
I don't think CRDs would typical share the same labels as the target pods. For now, I am ok with a namespace-wide query without labels. If we really want to allow them to target specific CRDs, it should be a separate config, not in the targetPodLabel list.

Reply:
With "namespace-wide" query you mean adding the "-n <namespace>" flag to the "oc get crds" command? That wouldn't work, because crds seem not to be namespaced. Their spec says whether the its CR objects can be namespaced or cluster-wide, but the CRD itself is always cluster wide. So I can see two options:
a) Use the "-l" to filter CRDs with custom labels for CRDs (different from the pod ones that I'm using in my PR) as you've suggested.
b) 1. Get a list of all CRDs of the cluster. 2. Check if there are CR objects of that CRD in the CNF target namespace. If so, fail if the that specific CRD has no status spec.

Reply:
ah, i see. maybe use spec.group instead of forcing them to label CRDs one by one. For option b, i worry if the CRs don't always exist.